tests/FSharp.Compiler.ComponentTests/Attributes/AttributePrivateSetterTests.fs

```diff
@@ -0,0 +1,44 @@
+namespace FSharp.Compiler.ComponentTests
+
+open Xunit
+open FSharp.Test.Compiler
+
+[<TestFixture>]
+type AttributePrivateSetterTests() =
+
+    [<Fact>]
+    let ``Compiler disallows setting a value of a private property on attribute``() =
+        // Arrange
+        let code = """
+using System;
+
+namespace AttributeTest
+{
+    [AttributeUsage(AttributeTargets.All)]
+    public class TestAttribute : Attribute
+    {
+        public TestAttribute(string name)
+        {
+            Name = name;
+        }
+
+        public string Name { get; }
+        
+        public bool IsDefault { get; private set; }
+    }
+}
+"""
+        let mainCode = """
+module AttributeTest
+
+[<AttributeTest.Test("TestName", IsDefault = true)>]
+type TestClass() = class end
+"""
+        // Act & Assert
+        FSharp mainCode
+        |> withAdditionalSources [(code, "cs")]
+        |> compile
+        |> shouldFail
+        |> withErrorCode "tcPropertyCannotBeSet1"
+        |> withSingleDiagnostic "Property 'IsDefault' on attribute (the setter is private) cannot be set"
```

Comment:
Instead of using `withErrorCode` + `withSingleDiagnostic`, it should match diagnostic with severity, line and column numbers.